src/qibolab/instruments/simulator/models/methods.py

```diff
@@ -0,0 +1,40 @@
+from typing import Union
+
+import yaml
+
+MHz = 1e6
+GHz = 1e9
+us = 1e-6
+ns = 1e-9
+
+
+def load_model_params(model_params: Union[dict, str]) -> dict:
+    """Load yaml to a dictionary or returns the input if it is already a
+    dictionary."""
+    if isinstance(model_params, dict):
+        params = model_params
+    else:
+        with open(model_params) as file:
+            params = yaml.safe_load(file)
+    return params
```

Comment:
I would suggest you to be as clear as possible about the inputs you expect: if you want a `dict`, just declare a `dict` and nothing else. If you want a path, ask for a `Path`.

Trying to bridge these gaps for the user generates a bunch of additional code, and it is more error-prone than just failing on incorrect types (sometimes it may succeed for the wrong reason, and then fail after).

Reply:
Ok will do. With the new arrangement to have `model.json`, the `Path` option is naturally preferred.

Reply:
Fixed in 76d5cb5